### PR TITLE
chore: recommend "env deploy" in "secret init"

### PR DESCRIPTION
**Pull request by efekarakus**
Also improve the recommended action when an env manifest isn't detected

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/cli/secret_init.go [hunk 1]
```diff
@@ -250,8 +257,8 @@ func (o *secretInitOpts) configureClientsAndUpgradeForEnvironments(secrets map[s
 		if err := o.configureClientsForEnv(envName); err != nil {
 			return err
 		}
-		if err := o.envUpgradeCMDs[envName].Execute(); err != nil {
-			return fmt.Errorf(`execute "env upgrade --app %s --name %s": %v`, o.appName, envName, err)
+		if err := validateMinEnvVersion(o.ws, o.envCompatibilityChecker[envName], o.appName, envName, "v1.4.0", "secret init"); err != nil {
```
Comment (dannyrandall) [inline internal/pkg/cli/secret_init.go]: nit: make `"v1.4.0"` a constant like `minEnvVersionSecretInit` or something

File: internal/pkg/cli/svc_deploy.go [hunk 2]
```diff
@@ -474,19 +480,40 @@ func (o *deploySvcOpts) getTargetApp() (*config.Application, error) {
...
 
 // RecommendActions returns recommended actions to be taken after the error.
 // Implements main.actionRecommender interface.
-func (e *errManifestIncompatibleWithEnvironment) RecommendActions() string {
-	return fmt.Sprintf("You can upgrade your environment template by running %s.\n", color.HighlightCode(fmt.Sprintf("copilot env deploy --name %s", e.envName)))
+func (e *errFeatureIncompatibleWithEnvironment) RecommendActions() string {
+	envs, _ := e.ws.ListEnvironments() // Best effort try to detect if env manifest exists.
+	for _, env := range envs {
+		if e.envName == env {
+			return fmt.Sprintf("You can upgrade your environment template by running %s.", color.HighlightCode(fmt.Sprintf("copilot env deploy --name %s", e.envName)))
```
Comment (paragbhingre) [inline internal/pkg/cli/svc_deploy.go]:
```suggestion
			return fmt.Sprintf("You can upgrade your %s environment template by running %s.", color.HighlightCode(fmt.Sprintf("copilot env deploy --name %s", e.envName, e.envName)))
```

How about adding an environment name in the prompt as well?

File: internal/pkg/cli/svc_deploy.go [hunk 3]
```diff
@@ -395,13 +397,17 @@ func validateManifestCompatibilityWithEnv(mft manifest.WorkloadManifest, envName
 			if v := template.LeastVersionForFeature(f); v != "" {
 				logMsg += fmt.Sprintf(` The least environment version that supports the feature is %s.`, v)
 			}
-			if currVersion, err := env.Version(); err == nil {
-				logMsg += fmt.Sprintf(" Your environment is on %s.", currVersion)
+			var currVersion string
+			if v, err := env.Version(); err == nil {
+				logMsg += fmt.Sprintf(" Your environment is on %s.", v)
+				currVersion = v
```
Comment (dannyrandall) [inline internal/pkg/cli/svc_deploy.go]:
```suggestion
			currVersion, err := env.Version()
			if err == nil {
				logMsg += fmt.Sprintf(" Your environment is on %s.", v)
```

File: internal/pkg/cli/svc_deploy.go [hunk 4]
```diff
@@ -474,19 +480,40 @@ func (o *deploySvcOpts) getTargetApp() (*config.Application, error) {
...
+	msgs := []string{
+		"You can upgrade your environment template by running:",
+		fmt.Sprintf("1. Create the directory to store your environment manifest %s.",
+			color.HighlightCode(fmt.Sprintf("mkdir -p %s", filepath.Join("copilot", "environments", e.envName)))),
+		fmt.Sprintf("2. Generate your environment manifest %s.",
+			color.HighlightCode(fmt.Sprintf("copilot env show -n %s --manifest > %s", e.envName, filepath.Join("copilot", "environments", e.envName, "manifest.yml")))),
+		fmt.Sprintf("3. Deploy your environment stack %s.",
+			color.HighlightCode(fmt.Sprintf("copilot env deploy --name %s", e.envName))),
+	}
+	return strings.Join(msgs, "\n")
```
Comment (paragbhingre) [inline internal/pkg/cli/svc_deploy.go:514]: This is neat 💯

**Review by dannyrandall — APPROVED** [review]
just a couple nits

File: internal/pkg/cli/svc_deploy.go (hunk 2 above)
Comment (efekarakus) [reply internal/pkg/cli/svc_deploy.go]: Done!

File: internal/pkg/cli/secret_init.go (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/cli/secret_init.go]: Done!

File: internal/pkg/cli/svc_deploy.go (hunk 3 above)
Comment (efekarakus) [reply internal/pkg/cli/svc_deploy.go]: Done!

File: internal/pkg/cli/svc_deploy.go [hunk 5]
```diff
@@ -474,19 +480,40 @@ func (o *deploySvcOpts) getTargetApp() (*config.Application, error) {
...
+		return fmt.Sprintf("environment %q is not on a version that supports the %q feature", e.envName, template.FriendlyEnvFeatureName(e.missingFeature))
+	}
+	return fmt.Sprintf("environment %q is on version %q which does not support the %q feature", e.envName, e.curVersion, template.FriendlyEnvFeatureName(e.missingFeature))
 }
 
 // RecommendActions returns recommended actions to be taken after the error.
 // Implements main.actionRecommender interface.
-func (e *errManifestIncompatibleWithEnvironment) RecommendActions() string {
-	return fmt.Sprintf("You can upgrade your environment template by running %s.\n", color.HighlightCode(fmt.Sprintf("copilot env deploy --name %s", e.envName)))
+func (e *errFeatureIncompatibleWithEnvironment) RecommendActions() string {
```
Comment (Lou1415926) [inline internal/pkg/cli/svc_deploy.go:498]: So noice

**Review by Lou1415926 — APPROVED** [review]

**Pull request merged**